pkg/metrics/metrics.go

```diff
@@ -38,7 +38,13 @@ func StartMetrics(config metricsConfig) error {
 	http.Handle(config.metricsPath, metricsHandler)
 	log.Info(fmt.Sprintf("Port: %s", config.metricsPort))
 	metricsPort := ":" + (config.metricsPort)
-	go http.ListenAndServe(metricsPort, nil)
+	go func() {
+		err := http.ListenAndServe(metricsPort, nil)
+		if err != nil {
+			// binding to port error
+			log.Error(err, "Binding to port failed")
```

Comment:
Should this error get returned from StartMetrics?

Reply:
Yes this lets us know from very early that binding failed. Is there another place you had in mind? Not sure I follow..

Reply:
You aren't returning this error anywhere